0004-ledger-plugin-interface/0004-ledger-interface.md

```diff
@@ -0,0 +1,203 @@
+# Ledger Interface
+
+This spec defines some guidelines for **RESTful ledger APIs** to easily integrate with the Interledger Protocol suite.
+
+The advantages of following this spec when designing a ledger are:
+- A standardized interface for inter-ledger communication
+- Easy adoption of existing interledger plugins
+
+## Requirements
+
+In order for ledgers to connect seamlessly with any of the ILP plugins `ilp-plugin-<ledgerID>`, the ledger needs to provide:
+- a RESTful HTTP API with endpoints to `transfer` (or `transaction`, `asset`) objects
+- PUSH notifications to register changes in the ledger objects, albeit using websockets, long-polling (EPOLL), or server-sent events.
+
+## RESTful API Methods
+
+### Ledger Metadata
+
+#### Retrieve Metadata
+
+##### Request
+```http
+GET / HTTP/1.1
+Host: ledger.my
+Authorization: <token>/<cert>/<password>/...
+Content-Type: application/json
+```
+
+##### Response
+```http
+HTTP/1.1 200 OK
+{
+    'precision': <int>,
+    'scale': <int>,
+    'currencyCode': <string: optional>,
+    'currencySymbol': <string: optional>
+}
+```
+
+### Accounts
+
+#### Retrieve account
+
+##### Request
+```http
+GET /accounts/<ID> HTTP/1.1
+Host: ledger.my
+Authorization: <token>/<cert>/<password>/...
+Content-Type: application/json
+```
+
+##### Response
+```http
+HTTP/1.1 200 OK
+{
+    'ledger': <uri>,
+    'name': <string: optional>,
+    'connector': <uri: optional>
+}
+```
+
+#### Update account
+
+##### Request
+```http
+PUT /accounts/<ID> HTTP/1.1
```

Comment:
The format for `<ID>` should be specified. Recommend an arbitrary string with characters `A-Z`, `a-z`, `0-9`, `-`, case-insensitive.